src/api/models/report.js

```diff
@@ -0,0 +1,43 @@
+import user from './user';
+
+export default (sequelize, DataTypes) => {
+  const Report = sequelize.define('Reports', {
+    userId: {
+      type: DataTypes.INTEGER,
+      references: {
+        model: 'Users',
+        key: 'id'
+      },
+      onDelete: 'CASCADE',
+      onUpdate: 'CASCADE',
+      allowNull: false
+    },
+    reportedArticleSlug: {
+      type: DataTypes.STRING,
+      references: {
+        model: 'Articles',
+        key: 'slug'
+      },
+      onDelete: 'CASCADE',
+      onUpdate: 'CASCADE',
+      allowNull: false
+    },
+    reasonId: {
+      type: DataTypes.INTEGER,
+      references: {
+        model: 'Reasons',
+        key: 'id'
+      },
+      onDelete: 'CASCADE',
+      onUpdate: 'CASCADE',
+      allowNull: false
+    }
+  });
+  Report.removeAttribute('id');
+  Report.associate = ({ User, Article, Reason }) => {
+    Report.belongsTo(User, { foreignKey: 'userId' });
+    Report.belongsTo(Article, { foreignKey: 'reportedArticleSlug' });
+    Report.belongsTo(Reason, { foreignKey: 'reasonId' });
```

Comment:
@MCFrank16 I understand why you've decided to use this association between Reason and Report, but I'd like to hear from you. Why not `has_one` and `belongs_to_many`?

Reply:
yeah, I think It would still work, but the ``belongs_to_many`` requires another table where the data passes ``through`` and it serves as a joint table with particularly the data from those 2 joined tables, so I tried to avoid many tasks and use the easiest way.